src/main/java/org/mockito/internal/progress/ThreadSafeMockingProgress.java

```diff
@@ -5,84 +5,32 @@
 
 package org.mockito.internal.progress;
 
-import org.mockito.internal.listeners.MockingProgressListener;
-import org.mockito.invocation.Invocation;
-import org.mockito.stubbing.OngoingStubbing;
-import org.mockito.verification.VerificationMode;
-import org.mockito.verification.VerificationStrategy;
-
-import java.io.Serializable;
-
-@SuppressWarnings("unchecked")
-public class ThreadSafeMockingProgress implements MockingProgress, Serializable {
-    
-    private static final long serialVersionUID = 6839454041642082618L;
-    private static final ThreadLocal<MockingProgress> mockingProgress = new ThreadLocal<MockingProgress>();
+/**
+ * Provides access to the {@link MockingProgress} of a corresponding {@link Thread}. Every {@link Thread} in Mockito has
+ * it s own {@link MockingProgress} to avoid data races while stubbing. 
+ */
+public class ThreadSafeMockingProgress {
 
-    static MockingProgress threadSafely() {
-        if (mockingProgress.get() == null) {
-            mockingProgress.set(new MockingProgressImpl());
+    private static final ThreadLocal<MockingProgress> SINGLETON = new ThreadLocal<MockingProgress>() {
```

Comment:
I see your point!  The name `THREAD_LOCAL_MOCKING_PROGRESS` is redundant to to the type `ThreadLocal<MockingProgress>`, it smells. What about  `MOCKING_PROGRESS_PROVIDER`  ?